lnd.go

```diff
@@ -186,6 +190,10 @@ type ListenerCfg struct {
 	// ExternalRestRegistrar is optional and specifies the registration
 	// callback to register external REST subservers.
 	ExternalRestRegistrar RestRegistrar
+
+	// BufListener is an optional field for passing in a grpc bufconn to
+	// listen for RPC requests on and make calls in-memory.
+	BufListener *ListenerWithSignal
```

Comment:
Why isn't the existing `RPCListener` enough here?

Reply:
Looks like this is actually just to pass the macaroon over? So the existing one can be used?

Reply:
Ah yeah we could. My reasoning for adding it as a separate field was mostly for clarity since we'll need both the buf and regular rpc listener & makes it easy to differentiate between them. But I suppose though we could just turn RPCListener into a slice and add both the regular listener and buf listener, then loop over the listeners inside, as needed? Is that what you're saying?

Reply:
Yeah, I think that's what would be best since we turn that into a slice later on anyway: https://github.com/lightningnetwork/lnd/blob/master/lnd.go#L327

Reply:
Sounds good, made that change